azurerm/resource_arm_subnet.go

```diff
@@ -335,10 +339,16 @@ func expandSubnetServiceEndpoints(d *schema.ResourceData) []network.ServiceEndpo
 func flattenSubnetServiceEndpoints(serviceEndpoints *[]network.ServiceEndpointPropertiesFormat) []string {
 	endpoints := make([]string, 0)
 
-	if serviceEndpoints != nil {
-		for _, endpoint := range *serviceEndpoints {
-			endpoints = append(endpoints, *endpoint.Service)
+	if serviceEndpoints == nil {
+		return endpoints
+	}
+
+	for _, endpoint := range *serviceEndpoints {
+		if endpoint.Service == nil {
```

Comment:
I think this would be better as:

```go
if v := endpoint.Service; v != nil {
  endpoints = append(endpoints, *endpoint.Service)
}
```

WDYT?

Reply:
The reason that sometimes I prefer to use `if-continue` is to reduce the nested if-else statements. For this case, it's OK to not use `if-continue`.